ibis/expr/operations/tests/test_generic.py

```diff
@@ -25,72 +25,52 @@ def test_literal_coercion_type_inference(value, dtype):
 
 
 def test_coerced_to_literal():
-    p = CoercedTo(ops.Literal)
     one = ops.Literal(1, dt.int8)
-    assert p.match(ops.Literal(1, dt.int8), {}) == one
-    assert p.match(1, {}) == one
-    assert p.match(False, {}) == ops.Literal(False, dt.boolean)
 
-    p = GenericCoercedTo(ops.Literal[dt.Int8])
-    assert p.match(ops.Literal(1, dt.int8), {}) == one
+    assert match(ops.Literal, 1) == one
+    assert match(ops.Literal, one) == one
+    assert match(ops.Literal, False) == ops.Literal(False, dt.boolean)
 
-    p = Pattern.from_typehint(ops.Literal[dt.Int8])
-    assert p == GenericCoercedTo(ops.Literal[dt.Int8])
+    assert match(ops.Literal[dt.Int8], 1) == one
+    assert match(ops.Literal[dt.Int16], 1) == ops.Literal(1, dt.int16)
```

Comment:
Can you avoid changing a bunch of code in a PR that's nominally only about reorganizing tests?

Reply:
Yes, I thought that it worth some simplification once I go trough the test cases.

Reply:
All those changes are semantically identical with the previous versions, just shorter in length and easier to read.

Reply:
Definitely in favor of the changes, just want to be able to give them the review they deserve!

Reply:
Would you like to see the changes in a separate commit or in a separate PR?

Reply:
Separate commit is fine!

Reply:
Done.